twisted/web/test/test_http2.py

```diff
@@ -1971,6 +1972,30 @@ def validate(streamID):
         return a._streamCleanupCallbacks[1].addCallback(validate)
 
 
+    def test_unnecessaryWindowUpdateForStream(self):
+        """
+        When a WindowUpdate frame is received for a stream but no data is
+        currently waiting, nothing exciting happens.
+        """
+        f = FrameFactory()
+        transport = StringTransport()
+        conn = H2Connection()
+        conn.requestFactory = DummyHTTPHandler
+
+        # Send the request.
+        frames = buildRequestFrames(
+            self.postRequestHeaders, self.postRequestData, f
+        )
+        frames.insert(1, f.buildWindowUpdateFrame(streamID=1, increment=5))
+        firstBytes = f.preamble()
+        firstBytes += b''.join(f.serialize() for f in frames[:2])
+
+        conn.makeConnection(transport)
+        conn.dataReceived(firstBytes)
+
+        self.assertRaises(priority.DeadlockError, next, conn.priority)
```

Comment:
I don't undertand how this relates to `nothing exciting happens` from the docstring .

I think that the docstring can be improved to be more specific. Since it is not `nothing happens` I assume that something will happen...just that it is not exciting :)

and with a better docstring, maybe this assertion will make sense